ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,210 @@
+    - [5.1.1 Approach 1 - clipboard-read permission required to listen to clipboardchange event](#511-approach-1---clipboard-read-permission-required-to-listen-to-clipboardchange-event)
+      - [Pros](#pros)
+      - [Cons](#cons)
+    - [5.1.2 Approach 2 - No permission required](#512-approach-2---no-permission-required)
+      - [Pros](#pros-1)
+      - [Cons](#cons-1)
+    - [5.1.3 Conclusion](#513-conclusion)
+  - [5.2 Page focus requirement](#52-page-focus-requirement)
+    - [5.2.1 Approach 1 - Page required to be in focus to receive event](#521-approach-1---page-required-to-be-in-focus-to-receive-event)
+      - [Pros](#pros-2)
+      - [Cons](#cons-2)
+    - [5.2.2 Approach 2 - No focus requirement](#522-approach-2---no-focus-requirement)
+      - [Pros:](#pros)
+      - [Cons:](#cons)
+    - [5.2.3 Conclusion](#523-conclusion)
+  - [5.3 Event details](#53-event-details)
+- [6. Detailed design discussion](#6-detailed-design-discussion)
+  - [6.1 Listen to clipboard change directly from the OS](#61-listen-to-clipboard-change-directly-from-the-os)
+      - [Pros:](#pros-1)
+      - [Cons:](#cons-1)
+    - [6.1.1 APIs provided by all OS to listen to clipboardchange event:](#611-apis-provided-by-all-os-to-listen-to-clipboardchange-event)
+  - [6.2 Considered alternative: Given the page focus restrictions, check clipboard hash change when page regains focus](#62-considered-alternative-given-the-page-focus-restrictions-check-clipboard-hash-change-when-page-regains-focus)
+    - [6.2.1 Platform support for getting clipboard SequenceNumber/Version number](#621-platform-support-for-getting-clipboard-sequencenumberversion-number)
+      - [Pros:](#pros-2)
+      - [Cons:](#cons-2)
+- [7. References & acknowledgements](#7-references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## 1. Introduction
+
+The clipboardchange event fires whenever the system clipboard contents are changed. This allows web-apps like remote desktop clients to stay in sync with the system clipboard. It provides an efficient alternative to polling the clipboard for changes.
+
+## 2. User scenarios
+
+### 2.1 Scenario: Sync clipboard with a remote desktop
+When a user copies text or an image on their local machine, the web-based remote desktop application can detect this clipboard change event through the browser's Clipboard API. Upon detecting the change, the application can automatically send the new clipboard content to the remote desktop environment.
+
+![](img/sync-clipboard-scenario.png)
+
+
+### 2.2 Scenario: Show available paste formats in web based editors
+Web based editors like Excel Online, Word Online may support paste operation in multiple formats. Within the UI, it may show the available formats like csv, image or plain text. The clipboard change event can be used to detect the change in available formats in clipboard and reflect the same on the UI as soon as it is changed. 
```

Comment:
Consider adding screenshots of the paste menu UI alongside the app names for Excel Online, Word Online, etc. That might make it easier for readers to grasp what is being described here.

Reply:
It might also be worth adding an example from a non-Microsoft web app, ex. Google Sheets, to showcase that this is a problem that broadly impacts the web.